aries_cloudagent/ledger/merkel_validation/domain_txn_handler.py

```diff
@@ -145,16 +144,15 @@ def prepare_get_attr_for_state(reply):
 
 def prepare_attr_for_state(txn, path_only=False):
     """Return key, value pair for state from ATTR."""
-    result = txn.get(RESULT)
-    txn_data = result.get(TXN).get(DATA)
+    txn_data = txn.get(TXN).get(DATA)
     nym = txn_data.get(DEST)
     attr_type, attr_key, value = parse_attr_txn(txn_data)
     path = make_state_path_for_attr(nym, attr_key, attr_type == HASH)
     if path_only:
         return path
     hashed_value = hash_of(value) if value else ""
-    seq_no = result.get(TXN_METADATA).get(SEQ_NO)
-    txn_time = result.get(TXN_METADATA).get(TXN_TIME)
+    seq_no = txn.get(TXN_METADATA).get(SEQ_NO)
```

Comment:
This pattern is used several times, but it's not as foolproof as it looks. If the first `get` fails then it will return None which doesn't have a `get` method.

Reply:
I think that `ATTRIB` write request reply will be structured as:
- `indy_sdk`
```
{
    'op': 'REPLY', 
    'result': {
        ...,
        "txn": {
           ...
        },
        "txnMetadata": {
            "seqNo": 10,  
            ...
        },
        ...
    }
}
```
- `indy_vdr`
```
{
    ...,
    "txn": {
       ...
    },
    "txnMetadata": {
        "seqNo": 10,  
        ...
    },
    ...
}
```
`prepare_attr_for_state` is not intended to be called directly and instead through `prepare_for_state_write` which should be able to handle this:
```
if RESULT in reply.keys():
   reply = reply.get(RESULT)
```
The same workflow happens with `prepare_for_state_read`.

Reply:
I'm just talking about the `.get(X).get(Y)`. It should be something like `.get(X, {}).get(Y)` or `[X].get(Y)` if the key is known to exist.

Reply:
Sounds good, I understand it now.